eth/stagedsync/stage_txlookup.go

```diff
@@ -0,0 +1,90 @@
+package stagedsync
+
+import (
+	"encoding/binary"
+	"errors"
+	"github.com/ledgerwatch/turbo-geth/common"
+	"github.com/ledgerwatch/turbo-geth/common/dbutils"
+	"github.com/ledgerwatch/turbo-geth/common/etl"
+	"github.com/ledgerwatch/turbo-geth/core/rawdb"
+	"github.com/ledgerwatch/turbo-geth/ethdb"
+	"github.com/ledgerwatch/turbo-geth/log"
+	"math/big"
+)
+
+func spawnTxLookup(s *StageState, db ethdb.Database, dataDir string, quitCh chan struct{}) error {
+	var blockNum uint64
+	var startKey []byte
+
+	lastProcessedBlockNumber := s.BlockNumber
+	if lastProcessedBlockNumber > 0 {
+		blockNum = lastProcessedBlockNumber + 1
+	}
+
+	startKey = dbutils.HeaderHashKey(blockNum)
+	err := TxLookupTransform(db, startKey, quitCh, dataDir, nil)
```

Comment:
there is now `s.UpdateWithStageData`, it stores the current load key, that you can later pass as `LoadStartKey` to ETL to make this interruptible
that we can hook into `etl`'s `OnLoadCommit`, see `TestTransformOnLoadCommitCustomBatchSize` test for how to use it.